modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/freelist/AbstractFreeList.java

```diff
@@ -267,6 +270,136 @@ private int addRowFragment(
         }
     }
 
+    /** Page handler that is used to update the expiration time of CacheDataRow. */
+    private final PageHandler<T, Boolean> updateDataRowTtl = new UpdateRowMetaHandler();
+
+    /**
+     * Page handler that is used to update the expiration time of CacheDataRow.
+     * This page handler is used only for CacheDataTree and PageIO is an instance of DataPageIO.
```

Comment:
Shouldn't it be in other package in this case? This one is called "abstract free list", please don't put implementation-specific functionality here

Reply:
Moved to CacheFreeList. I think this class is the most appropriate place.